File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.java
import java.util.*;

class Solution {
    public int solution(String t, String p) {
        int ans = 0;
        int pLength = p.length();
        long numP = Long.parseLong(p);

        for (int i = 0; i <= t.length() - pLength; i++) {
            if (Long.parseLong(t.substring(i, pLength + i)) <= numP) {
                ans++;
            }
        }
        return ans;
    }
}

File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.js
function solution(t, p) {
    let result = 0;
    let num = Number(p);
    for (let i = 0; i <= t.length - p.length; i++) {
        let tmp = t.substring(i, i + p.length);
        if (Number(tmp) <= num) {
            result++;
        }
    }
    return result;
}